manim/config/logger.py

```diff
@@ -1,120 +0,0 @@
-"""
-logger.py
----------
-This is the logging library for manim.
-This library uses rich for coloured log outputs.
-
-"""
-
-
-__all__ = ["logger", "console"]
-
-
-import configparser
-import logging
-
-from rich.console import Console
-from rich.logging import RichHandler
-from rich.theme import Theme
-from rich import print as printf
-from rich import errors, color
-import json
-import copy
-
-
-class JSONFormatter(logging.Formatter):
-    """Subclass of `:class:`logging.Formatter`, to build our own format of the logs (JSON)."""
-
-    def format(self, record):
-        record_c = copy.deepcopy(record)
-        if record_c.args:
-            for arg in record_c.args:
-                record_c.args[arg] = "<>"
-        return json.dumps(
-            {
-                "levelname": record_c.levelname,
-                "module": record_c.module,
-                "message": super().format(record_c),
-            }
-        )
-
-
-def _parse_theme(config_logger):
-    theme = dict(
-        zip(
-            [key.replace("_", ".") for key in config_logger.keys()],
-            list(config_logger.values()),
-        )
-    )
-
-    theme["log.width"] = None if theme["log.width"] == "-1" else int(theme["log.width"])
-
-    theme["log.height"] = (
-        None if theme["log.height"] == "-1" else int(theme["log.height"])
-    )
-    theme["log.timestamps"] = False
-    try:
-        customTheme = Theme(
-            {
-                k: v
-                for k, v in theme.items()
-                if k not in ["log.width", "log.height", "log.timestamps"]
-            }
-        )
-    except (color.ColorParseError, errors.StyleSyntaxError):
-        customTheme = None
-        printf(
-            "[logging.level.error]It seems your colour configuration couldn't be parsed. Loading the default color configuration...[/logging.level.error]"
-        )
-    return customTheme
-
-
-def set_rich_logger(config_logger, verbosity):
-    """Will set the RichHandler of the logger.
-
-    Parameter
-    ----------
-    config_logger :class:
-        Config object of the logger.
-    """
-    theme = _parse_theme(config_logger)
-    global console
-    console = Console(theme=theme)
-    # These keywords Are Highlighted specially.
-    RichHandler.KEYWORDS = [
-        "Played",
-        "animations",
-        "scene",
-        "Reading",
-        "Writing",
-        "script",
-        "arguments",
-        "Invalid",
-        "Aborting",
-        "module",
-        "File",
-        "Rendering",
-        "Rendered",
-    ]
-    rich_handler = RichHandler(
-        console=console, show_time=config_logger.getboolean("log_timestamps")
-    )
-    global logger
-    rich_handler.setLevel(verbosity)
-    logger.addHandler(rich_handler)
-
-
-def set_file_logger(log_file_path):
-    file_handler = logging.FileHandler(log_file_path, mode="w")
-    file_handler.setFormatter(JSONFormatter())
-    global logger
```

Comment:
Question : where did this go ? I couldn't find it in `config/utils.py`

Reply:
Currently, the only time at which the file logger is set is during `__main__.py`, so I moved this to `main_utils.py` (and refactored it a bit).

Reply:
I didn't see this file, sorry for that. But then why is `JsonFormatter` in `config/utils.py`? It is only used for the logs. Maybe you should move it to main_utils as well. 

Or separate logger stuff, I don't know.

Reply:
Yeah I'm not happy with how things turned out in terms of file organization.

Basically, `utils.py` contains everything needed by `__init__.py` to define things during library startup. Then `main_utils.py` contains everything needed by `__main__.py` to render a scene. I think that works but may not be the most logical way of separating things. And I think I missed JsonFormatter lol :sweat_smile: 

Don't worry, as I mentioned above, this is only one in a series of refactors, and my next PR already deals with these files.

Reply:
No problem then :)